launch_ros/launch_ros/actions/node.py

```diff
@@ -165,6 +187,20 @@ def _perform_substitutions(self, context: LaunchContext) -> None:
         if self.__expanded_node_namespace not in ['', '/']:
             self.__final_node_name += self.__expanded_node_namespace
         self.__final_node_name += '/' + self.__expanded_node_name
+        # expand parameters too
+        if self.__parameters is not None:
+            self.__expanded_parameters = []
+            for param_file_path in self.__parameters:
+                if isinstance(param_file_path, pathlib.Path):
+                    param_file_path = str(param_file_path)
+                expanded_param_file_path = perform_substitutions(
+                    context, normalize_to_list_of_substitutions(param_file_path))
+                if not os.path.isfile(expanded_param_file_path):
+                    _logger.warn(
+                        'Parameter file path is not valid: {}'.format(expanded_param_file_path))
```

Comment:
Would it be more accurate here to say "is not a file"? Is not valid implies a range of things to me, including that it exists but the contents are malformed.